### DISCUSSION TITLE
Improving debug experience

### DISCUSSION THREAD

**Discussion by davicostalf** [vtex/shoreline#731] [General]
Comentários no CSAT de Engenharia em 2023 H1:

> Bugs acontecem! Das poucas vezes que eu peguei, dá trabalho garantir que não foi um erro meu e depois muitas vezes eu vou pro próprio storybook do admin-ui tentar montar um exemplo para reproduzir o bug.  A mensagens de erro em si, eu não lembro de nenhuma específica, mas tenho a lembrança de serem úteis.

> Até agora todos os problemas que tive eu consegui encontrar a causa e/ou uma solução, também com o suporte do time. 
> Mas acredito nem sempre é muito fácil achar sozinho.

> The generated class names make it hard to match the HTML elements with the components and we don't have any support for something as emotion/styled-components have to generate class names based on the component name/file name.

**Comment by matheusps** [comment]
> The generated class names make it hard to match the HTML elements with the components and we don't have any support for something as emotion/styled-components have to generate class names based on the component name/file name. 

Não usamos mais css-in-js

> Bugs acontecem! Das poucas vezes que eu peguei, dá trabalho garantir que não foi um erro meu e depois muitas vezes eu vou pro próprio storybook do admin-ui tentar montar um exemplo para reproduzir o bug.  A mensagens de erro em si, eu não lembro de nenhuma específica, mas tenho a lembrança de serem úteis.

As mensagens de erro do Shoreline são mais significativas. Caso aparecem casos específicios, resolvemos caso a caso. Inclusive, esse comentário mostra a importância de ligarmos para o nosso storybook (que já estamos fazendo)

> Até agora todos os problemas que tive eu consegui encontrar a causa e/ou uma solução, também com o suporte do time.
Mas acredito nem sempre é muito fácil achar sozinho.

❤️
